docs/docs/unit-testing.md

```diff
@@ -0,0 +1,313 @@
+learn more about other tests that you can write.
+
+## Running tests
+
+If you look inside `package.json` you will probably find that there is already a
+script for `test`, which just outputs an error message. Change this to simply
+`jest`:
+
+```json
+  "scripts": {
+    "test": "jest"
+  }
+```
+
+This means you can now run tests by typing `npm run test`. If you want you could
+also add a script that runs `jest --watchAll` to watch files and run tests when
+they are changed.
+
+Now, run `npm run test` and you should immediately get an error like this:
+
+```sh
+ @font-face {
+    ^
+
+    SyntaxError: Invalid or unexpected token
+
+      2 |
+      3 | // Import typefaces
+    > 4 | import 'typeface-montserrat'
+```
+
+This is because the CSS mock doesn't recognize the `typeface-` modules. You can
+fix this easily by creating a new manual mock. Back in the `__mocks__`
+directory, create a file called `typeface-montserrat.js` and another called
+`typeface-merriweather.js`, each with the content `{}`. Any file in the mocks
+folder which has a name that matches that of a node_module is automatically used
+as a mock.
+
+Run the tests again now and it should all work! You should get a message about
+the snapshot being written. This is created in a `__snapshots__` directory next
+to your tests. If you take a look at it, you will see that it is a JSON
+representation of the `<Bio />` component. You should check your snapshot files
+into your SCM repository so that so that any changes are tracked in history.
+This is particularly important to remember if you are using a continuous
+integration system such as Travis to run tests, as these will fail if no
+snapshot is present.
+
+If you make changes that mean you need to update the snapshot, you can do this
+by running `npm run test -- -u`.
+
+## Using TypeScript
+
+If you are using TypeScript, you need to make a couple of small changes to your
+config. First install `ts-jest`:
+
+```sh
+npm install --save-dev ts-jest
+```
+
+Then edit the Jest config in your `package.json` to match this:
+
+```json
+  "jest": {
+    "transform": {
+        "^.+\\.tsx?$": "ts-jest",
+        "^.+\\.jsx?$": "<rootDir>/jest-preprocess.js"
+    },
+    "testRegex": "(/__tests__/.*\\.([tj]sx?)|(\\.|/)(test|spec))\\.([tj]sx?)$",
+    "moduleFileExtensions": [
+        "ts",
+        "tsx",
+        "js",
+        "jsx",
+        "json",
+        "node"
+    ],
+    "testPathIgnorePatterns": ["node_modules", ".cache"],
+    "transformIgnorePatterns": [
+      "node_modules/(?!(gatsby)/)"
+    ],
+    "globals": {
+      "__PATH_PREFIX__": ""
+    },
+    "setupFiles": [
+      "<rootDir>/loadershim.js"
+    ]
+  }
+```
+
+## Other potential issues
+
+If you are using the `Link` component then you may encounter this error when
+testing your component:
+
+```
+    TypeError: Cannot read property 'history' of undefined
+```
+
+This is an error in `gatsby-link`, although `react-router-dom` would complain
```

Comment:
This needs updated now that we've moved to @reach/router

- https://reach.tech/router/api/createMemorySource
- https://github.com/reach/router/issues/25#issuecomment-394003652

Reply:
OK, done. It simplifies it quite a bit, as `@reach/router`'s Link is a lot happier running in a test environment without a Router than react-router was.